### ISSUE TITLE
reconnect to BLENano

### ISSUE THREAD

**Issue by shokai** [shokai/blendmicro-node#9]
cannot reconnect on close

**Comment by shokai** [comment]
When call `ble.startAdvertising()` on BLENano, it advertises localname `DFU` only the first time. From next advertising, localname is changed to the specified name.

```c
static const uint8_t ble_name[] = "BLENano";

setup(){
  ble.init();
  ble.onDisconnection(disconnectionCallBack);
  ble.onDataWritten(writtenHandle);

  ble.accumulateAdvertisingPayload(GapAdvertisingData::BREDR_NOT_SUPPORTED);
  ble.accumulateAdvertisingPayload(GapAdvertisingData::SHORTENED_LOCAL_NAME,
                                   (const uint8_t *)ble_name, sizeof(ble_name) - 1);

  ble.setAdvertisingType(GapAdvertisingParams::ADV_CONNECTABLE_UNDIRECTED);
  ble.addService(service);
  ble.setDeviceName(ble_name);

  ble.setTxPower(TX_POWER);
  ble.setAdvertisingInterval(160);
  ble.setAdvertisingTimeout(0);
  ble.startAdvertising();
}
```

**Comment by shokai** [comment]
I have read around  `setAdvertisingData()` and `startAdvertising()` in `arduino-1.6.x/hardware/RBL/RBL_nRF51822/cores/RBL_nRF51822/BLE_API-master/ble/Gap.h`.

There are no suspicious part in BLENano. Probably this is mbed's bug.

**Comment by shokai** [comment]
Find another way of the `discover` event of `noble` npm to capture advertisement-packet.

**Comment by shokai** [comment]
For verification, write mbed BLE code without Arduino helpers then check changing of localName

**Comment by shokai** [comment]
Try to use 2nd argument of [noble](https://www.npmjs.com/package/noble) `startScanning`
```javascript
noble.startScanning([], true); // any service UUID, allow duplicates 
```

**Comment by shokai** [comment]
advertising DFU is a normal specification that enable [FOTA](https://github.com/RedBearLab/nRF51822-Arduino#ota) (Firmware on the air)

**Comment by shokai** [comment]
shipped v0.3.1
